### ISSUE TITLE
Add ActivityPub support to connect with PeerTube

### ISSUE THREAD

**Issue by Rudloff** [friendica/friendica#4592]
Hello,

[PeerTube](https://github.com/Chocobozzz/PeerTube) claims it is compatible with the Fediverse and can indeed interact with Mastodon.

But it seems Friendica doesn't know how to add a PeerTube user (`The network type couldn't be detected. Contact can't be added.`):
```
2018-03-12T21:36:53Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Probe.php:111:hostMeta	Probing for peertube.cpy.re
2018-03-12T21:36:53Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:Network.php:198:curl	https://peertube.cpy.re/.well-known/host-meta: 404 HTTP/1.1 404 Not Found
Server: nginx/1.10.3
Date: Mon, 12 Mar 2018 21:36:53 GMT
Transfer-Encoding: chunked
Connection: keep-alive
X-Powered-By: Express


2018-03-12T21:36:53Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Network.php:251:curl	error: https://peertube.cpy.re/.well-known/host-meta: 404 - 
2018-03-12T21:36:53Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:Network.php:252:curl	debug: Array
(
    [url] => https://peertube.cpy.re/.well-known/host-meta
    [content_type] => 
    [http_code] => 404
    [header_size] => 160
    [request_size] => 199
    [filetime] => -1
    [ssl_verify_result] => 0
    [redirect_count] => 0
    [total_time] => 0.214854
    [namelookup_time] => 0.060406
    [connect_time] => 0.092847
    [pretransfer_time] => 0.181765
    [size_upload] => 0
    [size_download] => 0
    [speed_download] => 0
    [speed_upload] => 0
    [download_content_length] => -1
    [upload_content_length] => -1
    [starttransfer_time] => 0.214827
    [redirect_time] => 0
    [redirect_url] => 
    [primary_ip] => 2a01:4f8:13b:20d6::99
    [certinfo] => Array
        (
        )

    [primary_port] => 443
    [local_ip] => 2a01:cb0c:3:1a00:1521:6f2f:fa32:95db
    [local_port] => 46696
)

2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:Network.php:198:curl	http://peertube.cpy.re/.well-known/host-meta: 301 HTTP/1.1 301 Moved Permanently
Server: nginx/1.10.3
Date: Mon, 12 Mar 2018 21:36:59 GMT
Content-Type: text/html
Content-Length: 185
Connection: keep-alive
Location: https://peertube.cpy.re/.well-known/host-meta
Strict-Transport-Security: max-age=15768000

<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.10.3</center>
</body>
</html>

2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:Network.php:198:curl	https://peertube.cpy.re/.well-known/host-meta: 404 HTTP/1.1 404 Not Found
Server: nginx/1.10.3
Date: Mon, 12 Mar 2018 21:36:59 GMT
Transfer-Encoding: chunked
Connection: keep-alive
X-Powered-By: Express


2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Network.php:251:curl	error: https://peertube.cpy.re/.well-known/host-meta: 404 - 
2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:Network.php:252:curl	debug: Array
(
    [url] => https://peertube.cpy.re/.well-known/host-meta
    [content_type] => 
    [http_code] => 404
    [header_size] => 160
    [request_size] => 199
    [filetime] => -1
    [ssl_verify_result] => 0
    [redirect_count] => 0
    [total_time] => 0.169493
    [namelookup_time] => 3.9E-5
    [connect_time] => 0.032811
    [pretransfer_time] => 0.136297
    [size_upload] => 0
    [size_download] => 0
    [speed_download] => 0
    [speed_upload] => 0
    [download_content_length] => -1
    [upload_content_length] => -1
    [starttransfer_time] => 0.169458
    [redirect_time] => 0
    [redirect_url] => 
    [primary_ip] => 2a01:4f8:13b:20d6::99
    [certinfo] => Array
        (
        )

    [primary_port] => 443
    [local_ip] => 2a01:cb0c:3:1a00:1521:6f2f:fa32:95db
    [local_port] => 46736
)

2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:XML.php:426:parseString	libxml: parse: error: 
2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Probe.php:131:hostMeta	No xrd object found for peertube.cpy.re
2018-03-12T21:36:59Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Probe.php:593:detect	No XRD data was found for root@peertube.cpy.re
2018-03-12T21:37:24Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Probe.php:1564:mail	searching root@peertube.cpy.re, 0 messages found.
2018-03-12T21:37:24Z@hsm0tai8fhdh50pqlmm8bnkg64	[DEBUG]:Profile.php:117:load	profile error: follow?url=root%40peertube.cpy.re&submit=Connect
2018-03-12T21:37:25Z@hsm0tai8fhdh50pqlmm8bnkg64	[DATA]:dba.php:960:delete	DELETE FROM `session` WHERE `expire` < 1520890645
```

(I have tested this with 4a0781db1.)

**Comment by MrPetovan** [comment]
This is because they support ActivityPub, which we are lagging to support ourselves.

**Comment by Lionirdeadman** [comment]
Shouldn't this be changed to "Support ActivityPub" and then we can test with let's say, Mastodon and Peertube? @MrPetovan

**Comment by MrPetovan** [comment]
Due to the vague nature of ActivityPub, I’m afraid we will need a whole new task to support Mastodon’s ActivityPub implementation.

**Comment by Lionirdeadman** [comment]
Yeah, I thought so.. Should we fuse them in here and have a tickbox for each or make another issue for Mastodon? 

I think fusing them would be better to not create needless amounts of issues but feel free to do whichever.

@MrPetovan

**Comment by MrPetovan** [comment]
I think it is better if we can track both tasks separately because they just are loosely related.

**Comment by Lionirdeadman** [comment]
Alrighty. I'll let you create the issue.

**Comment by strypey** [comment]
I'm not sure why @MrPetovan thinks ActivityPub is "vague" or that supporting two different ActivityPub-compatible server packages require tracking two separate issues. Sure if Friendica passes the [test suite](https://test.activitypub.rocks/), it should interoperate with any instance of any package using ActivityPub?

**Comment by MrPetovan** [comment]
This my utter wish, and at the same time I’m 100% convinced we’ll have some network-specific forks in our code.

Can I ask why do you care about the Friendica GitHub housekeeping policies?

**Comment by strypey** [comment]
@MrPetovan
> Can I ask why do you care about the Friendica GitHub housekeeping policies?

See my [comment on the Mastodon Issue](https://github.com/friendica/friendica/issues/4688#issuecomment-379459595).

**Comment by annando** [comment]
I don't know by now. It is possible that it works in some way, although we currently don't use the whole amount of the provided data.

**Comment by annando** [comment]
I did the same :-)

**Comment by annando** [comment]
There is currently a problem following peertube accounts. On the 35C3 we tried to analyse it and ended open up an issue: https://github.com/Chocobozzz/PeerTube/issues/1518

**Comment by MrPetovan** [comment]
Removing it from the milestone since it's depending on a thrid-party project.

**Comment by MrPetovan** [comment]
Closed via #6447
